services/catalog/src/simcore_service_catalog/api/routes/services.py

```diff
@@ -103,70 +123,33 @@ async def list_services(
                 inputs={},
                 outputs={},
             )
-            for key, version in visible_services
+            for key, version in services_in_db
         ]
         return services_overview
 
-    # Detailed view re-directing to
-
+    # NOTE: for the details of the services:
+    # 1. we get all the services from the director-v0 (TODO: move the registry to the catalog)
+    # 2. we filter the services using the visible ones
     # get the services from the registry and filter them out
-    frontend_services = list_frontend_services()
-    registry_services = await director_client.get("/services")
-    detailed_services_metadata = frontend_services + registry_services
-    detailed_services: List[ServiceOut] = []
-    for detailed_metadata in detailed_services_metadata:
-        try:
-            service_key, service_version = (
-                detailed_metadata.get("key"),
-                detailed_metadata.get("version"),
-            )
-            if (service_key, service_version) not in visible_services:
-                # no access to that service
-                continue
-
-            service = ServiceOut.parse_obj(detailed_metadata)
-
-            # Write Access Granted: fill in the service rights
-            access_rights: List[
-                ServiceAccessRightsAtDB
-            ] = await services_repo.get_service_access_rights(
-                service.key, service.version, product_name=x_simcore_products_name
-            )
-            service.access_rights = {rights.gid: rights for rights in access_rights}
-
-            # Write Access Granted: override some of the values with what is in the db
-            service_in_db: Optional[
-                ServiceMetaDataAtDB
-            ] = await services_repo.get_service(service.key, service.version)
-            if not service_in_db:
-                logger.error(
-                    "The service %s:%s is not in the database",
-                    service.key,
-                    service.version,
-                )
-                continue
+    filtered_services_in_registry = [
+        s
+        for s in list_frontend_services() + await director_client.get("/services")
+        if (s.get("key"), s.get("version")) in services_in_db
+    ]
 
-            service = service.copy(
-                update=service_in_db.dict(exclude_unset=True, exclude={"owner"})
-            )
-            # the owner shall be converted to an email address
-            if service_in_db.owner:
-                service.owner = await groups_repository.get_user_email_from_gid(
-                    service_in_db.owner
+    with ProcessPoolExecutor() as pool:
```

Comment:
shouldn't we have an executor pool __per application__ instead of __per request__? Did you test the difference?

Reply:
yes. saw nothing